File: rapid-router/src/main/java/com/codeforlife/rapidrouter/MainViewPagerAdapter.java
package com.codeforlife.rapidrouter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.PagerAdapter;

class MainViewPagerAdapter extends FragmentStatePagerAdapter {

    public MainViewPagerAdapter(FragmentManager manager) {
        super(manager);
    }

    @Override
    public Fragment getItem(int position) {
        if (position == 0) {
            return new BlocklyFragment();
        }
        return new GameFragment();
    }

    @Override
    public int getCount() {
        return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if (position == 0) {
            return "Blockly Stage";
        }
        return "Game View";
    }
}

File: rapid-router/src/main/java/com/codeforlife/rapidrouter/MainViewPagerAdapter.kt
package com.codeforlife.rapidrouter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

internal class MainViewPagerAdapter(manager: FragmentManager) : FragmentStatePagerAdapter(manager) {

    override fun getItem(position: Int): Fragment {
        return if (position == 0) {
            BlocklyFragment()
        } else GameFragment()
    }

    override fun getCount(): Int {
        return 2
    }

    override fun getPageTitle(position: Int): CharSequence {
        return if (position == 0) {
            "Blockly Stage"
        } else "Game View"
    }
}
